go-controller/pkg/ovn/pods_test.go

```diff
@@ -897,6 +897,46 @@ var _ = ginkgo.Describe("OVN Pod Operations", func() {
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 		})
 
+		ginkgo.It("remove a LSP from a pod that has no OVN annotations", func() {
+			app.Action = func(ctx *cli.Context) error {
+				namespaceT := *newNamespace("namespace1")
+				t := newTPod(
+					"node1",
+					"10.128.1.0/24",
+					"10.128.1.2",
+					"10.128.1.1",
+					"myPod",
+					"10.128.1.3",
+					"0a:58:0a:80:01:03",
+					namespaceT.Name,
+				)
+				pod := newPod(t.namespace, t.podName, t.nodeName, t.podIP)
+				fakeOvn.startWithDBSetup(initialDB,
+					&v1.NamespaceList{
+						Items: []v1.Namespace{
+							namespaceT,
+						},
+					},
+					&v1.PodList{
+						Items: []v1.Pod{
+							*pod,
+						},
+					},
+				)
+				annotations := getPodAnnotations(fakeOvn.fakeClient.KubeClient, t.namespace, t.podName)
+				gomega.Expect(annotations).To(gomega.Equal(""))
+
+				// Deleting port from a pod that has no annotations should be okay
+				err := fakeOvn.controller.deleteLogicalPort(pod, nil)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
```

Comment:
test looks good on first glance Flavio! qq: if you move this test into a new commit and your fix isn't there, this test will pass with `gomega.Expect(err).To(gomega.HaveOccurred())` right? as we except an error? and then in the second commit where you are doing the actual fix, you can simply change this line to `NotTo`. If you have already tested it that way its fine.

Reply:
ah, yes. That is actually how I tested it, @tssurya . Running this test w/out the changes in go-controller/pkg/ovn/pods.go will fail. You would like to see this test in a separate commit for checking it, or how we should merge this PR?

Reply:
ack if you have tested it that way I'm good as is. thanks flavio!